tests/e2e/vm_creation_performance_test.go

```diff
@@ -0,0 +1,233 @@
+					Name: cvmiName.Name,
+				},
+			},
+			PersistentVolumeClaim: v1alpha2.VMDPersistentVolumeClaim{
+				Size: resource.NewQuantity(6*1024*1024*1024, resource.BinarySI),
+			},
+		},
+	}
+	return vmd
+}
+
+func createVM(client kubeclient.Client, vm v1alpha2.VirtualMachine, vmd v1alpha2.VirtualMachineDisk, namespace string) {
+	resVMD, err := client.VirtualMachineDisks(namespace).Create(context.TODO(), &vmd, metav1.CreateOptions{})
+	Expect(err).NotTo(HaveOccurred(), "cannot create vmd - %s", &vmd.Name)
+	resVM, err := client.VirtualMachines(namespace).Create(context.TODO(), &vm, metav1.CreateOptions{})
+	Expect(err).NotTo(HaveOccurred(), "cannot create vm - %s", &vm.Name)
+	fmt.Println("VM", resVM.Name, "with VMD", resVMD.Name, "created")
+}
+
+var _ = Describe("Performance test 20 vm creation", Ordered, ContinueOnFailure, func() {
+	const (
+		vmName                         = "perf-test-vm"
+		vmdName                        = "perf-test-vmd"
+		cvmiName                       = "ubuntu-22.04"
+		vmCount                        = 26
+		notRunningVMCount              = 0
+		minutesLimits                  = 12 * time.Minute
+		deleteGracePeriodSeconds int64 = 30
+	)
+
+	var cvmi *v1alpha2.ClusterVirtualMachineImage
+	var vmdList []string
+	var vmMap []string
+
+	clientConfig := kubeclient.DefaultClientConfig(&pflag.FlagSet{})
+	client, err := kubeclient.GetClientFromClientConfig(clientConfig)
+	if err != nil {
+		log.Fatalf("Cannot obtain Virtualization client: %v\n", err)
+	}
+
+	AfterAll(func() {
+		By("Delete all resources")
+		for _, name := range vmMap {
+			err = client.VirtualMachines(conf.Namespace).Delete(context.TODO(), name, metav1.DeleteOptions{
+				GracePeriodSeconds: func(i int64) *int64 { return &i }(deleteGracePeriodSeconds)})
+			//Expect(err).NotTo(HaveOccurred())
+			if err != nil {
+				fmt.Println(err)
+			}
+		}
+		//for _, disk := range vmdList {
+		//	err = client.VirtualMachineDisks(conf.Namespace).Delete(context.TODO(), disk, metav1.DeleteOptions{
+		//		GracePeriodSeconds: func(i int64) *int64 { return &i }(deleteGracePeriodSeconds)})
+		//	//Expect(err).NotTo(HaveOccurred())
+		//	if err != nil {
+		//		fmt.Println(err)
+		//	}
+		//}
+	})
+
+	Context("VM", func() {
+		cvmi = CVMI(client, cvmiName, "create")
+		It("Create", func() {
+			for i := 1; i <= vmCount; i++ {
+				vmd := VMD(conf.Namespace, fmt.Sprintf("%s-%d", vmdName, i), *cvmi)
+				vm := VM(conf.Namespace, fmt.Sprintf("%s-%d", vmName, i), vmd)
+				createVM(client, vm, vmd, conf.Namespace)
+				vmMap = append(vmMap, vm.Name)
+				vmdList = append(vmdList, vmd.Name)
+			}
+		})
+		It("Wait until all virtual machines are running", func() {
+			runningVM := 0
+
+			start := time.Now()
+			fmt.Println("Starting at", start)
+
+			for len(vmMap) != notRunningVMCount {
+
+				for i, name := range vmMap {
+
+					vm, err := client.VirtualMachines(conf.Namespace).Get(context.TODO(), name, metav1.GetOptions{})
```

Comment:
We create VMs in one namespace, so it seems one List will be sufficient instead of multiple Gets